Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h

```diff
@@ -0,0 +1,817 @@
+                      bool useGroupWksp) {
+
+    setUp_HRP38692();
+
+    AlignAndFocusPowder align_and_focus;
+    align_and_focus.initialize();
+
+    align_and_focus.setPropertyValue("InputWorkspace", inputWS);
+    align_and_focus.setPropertyValue("OutputWorkspace", outputWS);
+    align_and_focus.setProperty("ResampleX", 1000);
+    align_and_focus.setProperty("Dspacing", false);
+
+    TS_ASSERT_THROWS_NOTHING(align_and_focus.execute());
+    TS_ASSERT(align_and_focus.isExecuted());
+
+    std::string calfilename("hrpd_new_072_01.cal");
+    if (useCalfile)
+      align_and_focus.setPropertyValue("CalFilename", calfilename);
+    else if (useCalWksp) {
+      loadDiffCal(calfilename, false, true, true);
+      align_and_focus.setPropertyValue("GroupingWorkspace",
+                                       loadDiffWSName + "_group");
+      align_and_focus.setPropertyValue("CalibrationWorkspace",
+                                       loadDiffWSName + "_cal");
+      align_and_focus.setPropertyValue("MaskWorkspace",
+                                       loadDiffWSName + "_mask");
+    }
+
+    if (useGroupfile)
+      align_and_focus.setPropertyValue("GroupFilename", calfilename);
+    else if (useGroupWksp) {
+      loadDiffCal(calfilename, true, false, true);
+      align_and_focus.setPropertyValue("MaskWorkspace",
+                                       loadDiffWSName + "_mask");
+      align_and_focus.setPropertyValue("GroupingWorkspace",
+                                       loadDiffWSName + "_group");
+    }
+
+    TS_ASSERT_THROWS_NOTHING(align_and_focus.execute());
+    TS_ASSERT(align_and_focus.isExecuted());
+
+    inWS = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(inputWS);
+    outWS =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(outputWS);
+
+    TS_ASSERT_EQUALS(inWS->size(), 263857);
+    TS_ASSERT_EQUALS(inWS->blocksize(), 23987);
+
+    TS_ASSERT_EQUALS(outWS->getAxis(0)->unit()->unitID(), "TOF");
+    TS_ASSERT_EQUALS(outWS->size(), 1000);
+    TS_ASSERT_EQUALS(outWS->blocksize(), outWS->size());
+    TS_ASSERT_EQUALS(outWS->getNumberHistograms(), 1);
+
+    // Maximum of peak near TOF approx. equal to 22,000 (micro-seconds)
+    TS_ASSERT_DELTA(outWS->x(0)[333], 21990.0502, 0.0001);
+    TS_ASSERT_DELTA(outWS->y(0)[333], 770.2515, 0.0001);
+
+    // Maximum of peak near TOF approx. equal to 25,800 (micro-seconds)
+    TS_ASSERT_DELTA(outWS->x(0)[398], 25750.3113, 0.0001);
+    TS_ASSERT_DELTA(outWS->y(0)[398], 1522.3778, 0.0001);
+
+    // Maximum of peak near TOF approx. equal to 42,000 (micro-seconds)
+    TS_ASSERT_DELTA(outWS->x(0)[600], 42056.6091, 0.0001);
+    TS_ASSERT_DELTA(outWS->y(0)[600], 7283.29652, 0.0001);
+  }
+
+  /* Setup for event data */
+
+  void setUp_EventWorkspace(int numBanks, int numPixels,
+                            int numEvents = 10000) {
+    inputWS = "eventWS";
+
+    CreateSampleWorkspace createSampleAlg;
```

Comment:
It would be a good idea (follow-on PR) to make this the default functionality for `CreateSampleWorkspace(Function='Powder Diffraction', ...)` as having the instrument centered on two_theta=90 is fairly common.

Reply:
Issue created #24411